Automattic/jetpack
# Jetpack AI: fix site-require-upgrade field to be ready for tiered plans

Part of #33824.

## Proposed changes:

* Remove the check for the AI Assistant feature presence on the `site-require-upgrade` field, so even when the site has the feature, it may require an upgrade when it's over the usage limit

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Let's test if the flag is working correctly for the current plan structure (free and unlimited paid)
* Apply this branch change to your sandbox and sandbox the `public-api` endpoint
* Go to the WordPress.com console to execute requests to `wpcom/v2` `/sites/$wpcom_site/jetpack-ai/ai-assistant-feature`
* Check the `site-require-upgrade` field on the response for some test scenarios:
   * Site with free plan and less than 20 requests -> `site-require-upgrade: false`
   * Site with free plan and more than 20 requests -> `site-require-upgrade: true`
   * Site with the paid plan -> `site-require-upgrade: false`
   * Internal P2 -> `site-require-upgrade: false`
   * Simple site with upgrade -> `site-require-upgrade: false`
* If you have a site with a tiered plan attached to it, test using it and confirm the field will be false when you are under the limit, and true when you are over the limit
